python/lsst/cp/pipe/cpBfk.py

```diff
@@ -415,7 +430,52 @@ def run(self, inputPtc, dummy, camera, inputDims):
             if self.config.level == 'DETECTOR' and ampName not in self.config.ignoreAmpsForAveraging:
                 detectorCorrList.extend(scaledCorrList)
                 detectorFluxes.extend(fluxes)
-            bfk.valid[ampName] = True
+
+            # Check for validity
+            if self.config.doCheckValidity:
+                kernelSize = bfk.ampKernels[ampName].shape[0]
+                kernelCenterValue = bfk.ampKernels[ampName][kernelSize//2, kernelSize//2]
+                xv, yv = np.meshgrid(range(kernelSize), range(kernelSize))
+
+                # Get a mask for a 3px picture frame
+                valid = True
+                if kernelSize >= 7:
+                    edges = (xv < 3)
+                    edges += (xv > kernelSize - 3 - 1)
+                    edges += (yv < 3)
+                    edges += (yv > kernelSize - 3 - 1)
+
+                    kernelEdges = bfk.ampKernels[ampName][edges].ravel()
+                    kernelEdgeStd = np.std(kernelEdges)
+
+                    # Edges should converge to zero
+                    valid = np.all(np.isclose(
+                        kernelEdges, 0,
+                        rtol=self.config.nSigmaTolForValidEdge*kernelEdgeStd,
+                    ))
+                    if not valid:
+                        self.log.info(f"{ampName}: the kernel edges did not converge to 0 within "
+                                      f"+/- {self.config.nSigmaTolForValidEdge*kernelEdgeStd} sigma")
+
+                    # Kernel should be negative
+                    valid *= np.all(bfk.ampKernels[ampName][~edges] <= 5*kernelEdgeStd)
+                    if not np.all(bfk.ampKernels[ampName][~edges] <= 5*kernelEdgeStd):
+                        self.log.info(f"{ampName}: the kernel is not negative")
+
+                    # The center should be the minimum
+                    valid *= np.all(bfk.ampKernels[ampName] >= kernelCenterValue)
+                    if not np.all(bfk.ampKernels[ampName] >= kernelCenterValue):
+                        self.log.info(f"{ampName}: the kernel center is not the absolute minimum.")
+                else:
+                    self.log.info(f"{ampName}: The kernel is too small for validity check. Setting "
+                                  "bfk.valid=False.")
```

Comment:
This is a config failure, isn't it?  This should be checked in the config validation and not run if you combine these settings.  Basically a RuntimeError not an info log.